### PR TITLE
Update snowflake hook to not use extra prefix

### PR DESCRIPTION
**Pull request by dstandish**
As of Airflow 2.3 it's no longer necessary to use the `extra__<conn_type>__` prefix for connection extras.

We're bumping min airflow version to 2.3 now, so we can remove the prefix.

We handle back compat with method get_field.

And in version 2.5 we will support no prefix in `get_ui_field_behaviors` (pr https://github.com/apache/airflow/pull/26995), so we prepare for that here.  What we do is update the placeholders to remove the prefix but add in the logic (which in 2.5 is added to providers manager) to add the prefix.  This way all we need to do when min airflow version is 2.5 is remove the decorator.  And in the meantime, we don't ever have to see the prefix.

### CODE REVIEW

File: airflow/providers/snowflake/hooks/snowflake.py
Comment (eladkal) [inline airflow/providers/snowflake/hooks/snowflake.py]: Next provider release cycle is going to set min Airflow version 2.3 for all providers:
https://github.com/apache/airflow#release-process-for-providers

Since the next cycle is on October any way maybe we should hold this PR till we adjust the min Airflow version?

Comment (dstandish) [reply airflow/providers/snowflake/hooks/snowflake.py]: sure, that's fine

Comment (dstandish) [reply airflow/providers/snowflake/hooks/snowflake.py]: when do you think we will bump the min version?  will there be time to add this after bumping and before release?

Comment (eladkal) [reply airflow/providers/snowflake/hooks/snowflake.py]: The bump can happen only after 11-Oct because we might need to make ad-hoc release to fix something urgently. Right after 11-Oct we can bump all providers and then proceed.

The reason I suggested it is because we usually do 1 release per month so the next release is expected around the end of October thus the current "backport" code for 2.2 will never reach to the user anyway.

We can also merge this first and do cleanup after.. That is OK.

File: airflow/providers/snowflake/hooks/snowflake.py
Comment (bharanidharan14) [inline airflow/providers/snowflake/hooks/snowflake.py]: Can we check for the exact string `extra__` . WDYT ?

File: airflow/providers/snowflake/hooks/snowflake.py
Comment (bharanidharan14) [inline airflow/providers/snowflake/hooks/snowflake.py:199]: we can remove `or None`. WDYT

**Review by eladkal — APPROVED** [review]

**Pull request merged**
